firmware/common2015/drivers/cc1201/CC1201Config.cpp

```diff
@@ -0,0 +1,1212 @@
+    device->writeReg(CC1201EXT_CHFILT_Q1, config->chfiltQ1, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_CHFILT_Q0, config->chfiltQ0, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_GPIO_STATUS, config->gpioStatus, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_FSCAL_CTRL, config->fscalCtrl, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_PHASE_ADJUST, config->phaseAdjust, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_PARTNUMBER, config->partnumber, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_PARTVERSION, config->partversion, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_SERIAL_STATUS, config->serialStatus,
+    EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_MODEM_STATUS1, config->modemStatus1,
+    EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_MODEM_STATUS0, config->modemStatus0,
+    EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_MARC_STATUS1, config->marcStatus1, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_MARC_STATUS0, config->marcStatus0, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_PA_IFAMP_TEST, config->paIfampTest, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_FSRF_TEST, config->fsrfTest, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_PRE_TEST, config->preTest, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_PRE_OVR, config->preOvr, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_ADC_TEST, config->adcTest, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_DVC_TEST, config->dvcTest, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_ATEST, config->atest, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_ATEST_LVDS, config->atestLvds, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_ATEST_MODE, config->atestMode, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_XOSC_TEST1, config->xoscTest1, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_XOSC_TEST0, config->xoscTest0, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_AES, config->aes, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_MDM_TEST, config->mdmTest, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_RXFIRST, config->rxfirst, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_TXFIRST, config->txfirst, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_RXLAST, config->rxlast, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_TXLAST, config->txlast, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_NUM_TXBYTES, config->numTxbytes, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_NUM_RXBYTES, config->numRxbytes, EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_FIFO_NUM_TXBYTES, config->fifoNumTxbytes,
+    EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_FIFO_NUM_RXBYTES, config->fifoNumRxbytes,
+    EXT_FLAG_ON);
+    device->writeReg(CC1201EXT_RXFIFO_PRE_BUF, config->rxfifoPreBuf,
+    EXT_FLAG_ON);
+    */
+    // TODO (if necessary) implement AES coverage (be same as above, just need
+    // to look at data sheet)
+
+    return device;
+}
+
+/**
+ * Verifies the configuration readback of the given CC1201 device against the
+ * given CC1201Config specification.
+ *
+ * @param CC1201Config* the instance holding the preferred CC1201 configuration
```

Comment:
Let's move all of the doxygen out of the .cpp files and into the corresponding .hpp.  It's best not to have two copies or else things get out of sync.